src/task/CreateSDPTask.ts

```diff
@@ -29,7 +29,7 @@ export default class CreateSDPTask extends BaseTask {
   }
 
   sessionUsesAudio(): boolean {
-    return true;
+    return !!this.context.meetingSessionConfiguration?.urls?.audioHostURL?.length;
```

Comment:
+1 for adding the new boolean flag. Also, should we set `audioCheckin` to this new boolean value when we do the signaling client subscription instead of hardcoding it false? https://github.com/aws/amazon-chime-sdk-js/blob/7a73e038bdea14c6339041e53d50d5d0de069d43/src/task/SubscribeAndReceiveSubscribeAckTask.ts#L79

Reply:
Is this addressed? Also, we are repeating `this.context.meetingSessionConfiguration?.urls?.audioHostURL?.length` multiple times. Would it be better to have a boolean flag in the [`AudioVideoControllerState`](https://github.com/aws/amazon-chime-sdk-js/blob/master/src/audiovideocontroller/AudioVideoControllerState.ts) itself?

Reply:
You've marked this as Resolved, but I still see a length check here…

Reply:
Oopsie, i think i replied in a wrong place.
I am slightly hesitant to blow up the mutable `AudioVideoControllerState` even more which is not well documented and there are a bunch of redundant states.

Reply:
At the very least I would suggest using `!!this.context.meetingSessionConfiguration?.urls?.audioHostURL` as your condition, not the length — an empty string is falsy.

Reply:
sounds good, changed the condition for all places.